### PR TITLE
Update product-configure schema to v3.4

### PR DESCRIPTION
**Pull request by james-smith-za**
Checklist (if not applicable, edit to add `(N/A)` and mark as done):

- [x] (N/A) If dependencies are added/removed: update `setup.cfg` and `.pre-commit-config.yaml`
- [x] (N/A) If modules are added/removed: use sphinx-apidoc to update files in `doc/`
- [x] Ensure copyright notices are present and up-to-date
- [x] (N/A) If qualification tests are changed: attach a sample qualification report
- [x] (N/A) If design has changed: ensure documentation is up to date
- [x] (N/A) If ICD-defined sensors have been added: update `fake_servers.py` in katsdpcontroller to match

Contributes to NGC-914.

### CODE REVIEW

File: scratch/sim_correlator.py [hunk 1]
```diff
@@ -133,17 +133,17 @@ def generate_config(args: argparse.Namespace) -> dict:
...
+        "src-streams": ["antenna-channelised-voltage"],
         "int_time": args.int_time,
     }
     if args.last_stage == "x":
         return config
 
     config["outputs"]["sdp_l0"] = {
         "type": "sdp.vis",
-        "src_streams": ["baseline_correlation_products"],
+        "src-streams": ["baseline-correlation-products"],
```
Comment (bmerry) [inline scratch/sim_correlator.py:146]:
```suggestion
        "src_streams": ["baseline-correlation-products"],
```

File: scratch/sim_correlator.py [hunk 2]
```diff
@@ -133,17 +133,17 @@ def generate_config(args: argparse.Namespace) -> dict:
     if args.last_stage == "f":
         return config
 
-    config["outputs"]["baseline_correlation_products"] = {
+    config["outputs"]["baseline-correlation-products"] = {
         "type": "gpucbf.baseline_correlation_products",
-        "src_streams": ["antenna_channelised_voltage"],
+        "src-streams": ["antenna-channelised-voltage"],
```
[the remark this answers is not in the document]
Comment (james-smith-za) [reply scratch/sim_correlator.py:138]: Oops.

**Comment by james-smith-za** [comment]
> Apart from the line-by-line changes, there are places in the test where `?delays` or `?gains` requests need to be updated.

I'd forgotten about that. The thought occurs to me now to wonder whether it's worthwhile defining a `F`/`XBENG_STREAM_NAME` variable or something like that.

**Comment by bmerry** [comment]
> > Apart from the line-by-line changes, there are places in the test where `?delays` or `?gains` requests need to be updated.
> 
> I'd forgotten about that. The thought occurs to me now to wonder whether it's worthwhile defining a `F`/`XBENG_STREAM_NAME` variable or something like that. I mean we shouldn't be changing it too often. But it may help make intent clearer and avoid bugs.

If we get it wrong, the test will fail, so I think the only advantage would be to catch bugs *earlier* (with mypy, instead of when the test runs). If you call it FENG_STREAM_NAME you're creating new problems:
1. It doesn't adhere to the system-wide naming convention, which is antenna-channelised-voltage.
2. There can be more than one stream coming out of an F-engine (and definitely more than one type coming out of an XB-engine).

**Comment by james-smith-za** [comment]
Will run a qualification test once Jenkins has finished building Docker images.

**Comment by james-smith-za** [comment]
It looks like it works.
[Uploading report.pdf…]()

**Review by bmerry — APPROVED** [review]

**Pull request merged**